info/v1/machine_test.go

```diff
@@ -0,0 +1,93 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMachineInfoClone(t *testing.T) {
+	machineInfo := getFakeMachineInfo()
+
+	// Make sure all fields are populated
+	machineInfoReflection := reflect.ValueOf(machineInfo)
+	for i := 0; i < machineInfoReflection.NumField(); i++ {
+		assert.Falsef(t, machineInfoReflection.Field(i).IsZero(), "field %s is not populated", machineInfoReflection.Type().Field(i).Name)
+	}
+
+	clonedMachineInfo := *machineInfo.Clone()
+	assert.Equal(t, machineInfo, clonedMachineInfo, "cloned machine info should be equal to the original")
+}
+
+func getFakeMachineInfo() MachineInfo {
+	return MachineInfo{
```

Comment:
nit: I wonder if it can be improved even further to fail if there are any field that we didn't populate. Even if it is a hardcoded list that is compared using reflection.

If this will introduce too much complexity, it is not worth it

Reply:
Hey @SergeyKanzhelev,

If I understand you correctly, this is exactly what I've done [here](https://github.com/google/cadvisor/pull/3293/files#diff-f5e724253acf06e5b358065792d006eaf93ceae521d8cd1c6b06306a926acd45R28).

Basically I've used reflection to check if there's a field that's populated with a "zero-value". This way if another field is added and isn't initialized in `getFakeMachineInfo()` the test would fail. See [this comment](https://github.com/google/cadvisor/pull/3293#issuecomment-1549229681) for more info.

Did I get you correctly?

Reply:
cool. sorry I missed it

Reply:
No worries!